lib/utils/report.mjs

```diff
@@ -16,54 +16,45 @@ import { isNumber } from './validateTypes.mjs';
  * @type {import('stylelint').Utils['report']}
  */
 export default function report(problem) {
-	const { ruleName, result, message, messageArgs, line, node, index, endIndex, word, severity } =
-		problem;
-
-	if (isFixApplied(problem)) return;
-
-	const ruleSeverityOption = severity || result.stylelint.ruleSeverities[ruleName];
-	const defaultSeverity = result.stylelint.config?.defaultSeverity || 'error';
-	const ruleSeverity =
-		typeof ruleSeverityOption === 'function'
-			? ruleSeverityOption(...(messageArgs || [])) || defaultSeverity
-			: ruleSeverityOption;
+	const { node, index, endIndex, line, start, end, result, ruleName, word, ...rest } = problem;
+	const {
+		disabledRanges,
+		quiet,
+		ruleSeverities,
+		config: { defaultSeverity = 'error', ignoreDisables } = {},
+		customMessages: { [ruleName]: message = rest.message } = {},
+		customUrls: { [ruleName]: customUrl } = {},
+	} = result.stylelint;
+	const { messageArgs = [], severity = ruleSeverities[ruleName] } = rest;
+	const ruleSeverity = isFn(severity) ? severity(...messageArgs) || defaultSeverity : severity;
 
 	// In quiet mode, mere warnings are ignored
-	if (result.stylelint.quiet && ruleSeverity !== 'error') {
-		return;
-	}
-
-	// endIndex is optional
-	const { start } = (node && node.rangeBy({ index, endIndex })) || {};
+	if (quiet && ruleSeverity !== 'error') return;
 
+	const range = node?.rangeBy({ index, endIndex, word }) ?? {};
```

Comment:
@Mouvedia Can you elaborate on this change?
Why is `word` included here?

I am fairly certain that this is a breaking change.

See : https://github.com/postcss/postcss/blob/f9a9868eb3a591cd99d504ac57bf7ee77380856b/lib/node.js#L222-L229

Reply:
#7779